### DISCUSSION TITLE
Update Custom Fields Through API

### DISCUSSION THREAD

**Discussion by Dnny44** [amidaware/tacticalrmm#879] [Ideas]
Would be nice to update Custom Fields through the API or a server based Python Script. Just mentioning so that it might end up on the roadmap one day! Fantastic project and great work!

**Comment by wh1te909** [comment]
our entire app is just one big API. frontend and backend are separate. so anything that can be done from the web UI, can also be done from a script using the API. just need to hit the custom field endpoint with the correct params

**Comment by Dnny44** [comment]
Thank @wh1te909, 30 minutes in Postman got it sorted. Just incase someone else is curious:

The "minimum" parameters for a POST request:

```
{
    "client": {
        "id": %clientID%
    },
    "custom_fields": [
        {
            "string_value": "Value1",
            "field": 1
        },
        {
            "string_value": "Value2",
            "field": 2
        }
    ]
}
```

It appears that you do not "need" the client section if you have assigned values to the custom fields before, but if you havent you will need it. Your endpoint would be %baseUrl%/clients/%clientID%.

The field ID's are not the same as the custom field ID but can be found by doing a get request at the client level to determine what they are.
